monero-sys/src/bridge.h

```diff
@@ -122,13 +122,63 @@ namespace Monero
         return wallet.createTransaction(dest_address, "", Monero::optional<uint64_t>(amount), 0, PendingTransaction::Priority_Default);
     }
 
+    /**
+     * Create a transaction that spends all the unlocked balance to a single destination.
+     */
     inline PendingTransaction *createSweepTransaction(
         Wallet &wallet,
         const std::string &dest_address)
     {
         return wallet.createTransaction(dest_address, "", Monero::optional<uint64_t>(), 0, PendingTransaction::Priority_Default);
     }
 
+    inline uint64_t estimateTransactionFee(
+        const Wallet &wallet,
+        uint64_t num_outputs)
+    {
+        std::vector<std::pair<std::string, uint64_t>> fee_dests;
+        for (uint64_t i = 0; i < num_outputs; ++i)
```

Comment:
Wrong, we need to:
- ensure balance (in piconeros is > num_outputs)
- drop the last address because wallet2 will create a change output. The change output will only be dropped if there is no XMR left after subtracting the fee